nuxeo-importer-stream/src/main/java/org/nuxeo/importer/stream/consumer/watermarker/JpegWatermarker.java

```
@@ -0,0 +1,78 @@
+import org.apache.commons.imaging.formats.tiff.TiffImageMetadata;
+import org.apache.commons.imaging.formats.tiff.constants.ExifTagConstants;
+import org.apache.commons.imaging.formats.tiff.write.TiffOutputDirectory;
+import org.apache.commons.imaging.formats.tiff.write.TiffOutputSet;
+import org.apache.commons.io.FileUtils;
+
+/**
+ * @since 10.2
+ */
+public class JpegWatermarker extends AbstractWatermarker {
+
+    @Override
+    public Path addWatermark(Path inputFile, Path outputDir, String watermark) {
+        File jpegImageFile = inputFile.toFile();
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        try {
+            TiffOutputSet outputSet = null;
+            ImageMetadata metadata = Imaging.getMetadata(jpegImageFile);
+            JpegImageMetadata jpegMetadata = (JpegImageMetadata) metadata;
+            if (null != jpegMetadata) {
+                TiffImageMetadata exif = jpegMetadata.getExif();
+                if (null != exif) {
+                    outputSet = exif.getOutputSet();
+                }
+            }
+            if (null == outputSet) {
+                outputSet = new TiffOutputSet();
+            }
+            final TiffOutputDirectory exifDirectory = outputSet.getOrCreateRootDirectory();
+            exifDirectory.removeField(ExifTagConstants.EXIF_TAG_SOFTWARE);
+            exifDirectory.add(ExifTagConstants.EXIF_TAG_SOFTWARE, watermark);
+            new ExifRewriter().updateExifMetadataLossless(jpegImageFile, os, outputSet);
+            os.flush();
+            File output = getOutputPath(inputFile, outputDir, watermark).toFile();
+            FileUtils.writeByteArrayToFile(output, os.toByteArray());
+            return output.toPath();
+        } catch (ImageReadException | ImageWriteException | IOException e) {
+            throw new IllegalArgumentException("Unable to edit jpeg " + inputFile, e);
+        } finally {
+            try {
+                os.close();
```

Use try-with-resources instead.

---

try-with-resources prevents to discard exception on close